### PR TITLE
Adds Dockerfile to build docker and/or singularity containers

### PR DESCRIPTION
**Pull request by daniel-thom**
Completes JIRA Story DSGRID-187.

I tested this out on Eagle with a 2-node Spark cluster. @lixiliu If you'd like to test this on Eagle or AWS before approving, that would be a good check.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+COPY docker/vimrc $HOME/.vimrc
+COPY docker/vimrc /data/vimrc
+
+RUN echo "$VERSION" > /repos/version.txt
+
+RUN --mount=type=secret,id=access_token ACCESS_TOKEN=$(cat /run/secrets/access_token) \
+    && git config --global url."https://${ACCESS_TOKEN}:@github.com/".insteadOf "https://github.com/"
+
+WORKDIR /repos
+# TODO: Find out how to install proper certificates.
```
Comment (elainethale) [inline Dockerfile:45]: Does the buildstockbatch Dockerfile provide any clues about certificates:

```
FROM nrel/openstudio:2.9.1

ENV LANG=C.UTF-8 LC_ALL=C.UTF-8

RUN sudo apt update && \
    sudo apt install -y wget build-essential checkinstall libreadline-gplv2-dev libncursesw5-dev libssl-dev \
    libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev libffi-dev zlib1g-dev

RUN wget https://www.python.org/ftp/python/3.8.8/Python-3.8.8.tgz && \
    tar xzf Python-3.8.8.tgz && \
    cd Python-3.8.8 && \
    ./configure --enable-optimizations && \
    make altinstall && \
    rm -rf Python-3.8.8 && \
    rm -rf Python-3.8.8.tgz

RUN sudo apt install -y -V ca-certificates lsb-release && \
    wget https://apache.bintray.com/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb && \
    sudo apt install -y -V ./apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb && \
    sudo apt update && \
    sudo apt install -y -V libarrow-dev libarrow-glib-dev libarrow-dataset-dev libparquet-dev libparquet-glib-dev

COPY . /buildstock-batch/
RUN python3.8 -m pip install /buildstock-batch
```

?

**Review by elainethale — COMMENTED** [review]
Maybe let's try it out during our meeting today?

File: Dockerfile (hunk 1 above)
Comment (daniel-thom) [reply Dockerfile:45]: Earlier I checked with Google and the answer was `ca-certifications`, which is above, but it didn't work. I'll explore a bit more.

Comment (daniel-thom) [reply Dockerfile]: > Earlier I checked with Google and the answer was `ca-certificates`, which is above, but it didn't work. I'll explore a bit more.

The error was caused by being on the VPN.

**Comment by lixiliu** [comment]
I ran an enumeration script (`projects/dsgrid/efs_datasets/enumeration/enumerate_load_data_table_lookup.py`) using the singularity image `/projects/dsgrid/containers/dsgrid. (singularity)` and it works well.

**Pull request merged**